test/NuGet.Clients.FuncTests/NuGet.CommandLine.FuncTest/Commands/PushCommandTest.cs

```diff
@@ -245,6 +256,512 @@ public void PushCommand_Server_SkipDuplicate_IsSpecified_PushProceeds()
+
+                    string expectedFileNotFoundErrorMessage = string.Format(MESSAGE_FILE_DOES_NOT_EXIST, snupkgToPush);
+
+                    Assert.True(1 == result.Item1, "File did not exist and should fail.");
+                    Assert.DoesNotContain(MESSAGE_PACKAGE_PUSHED, result.Item2);
+                    Assert.Contains(expectedFileNotFoundErrorMessage, result.Item3);
+                }
+            }
+        }
+
+        /// <summary>
+        /// When pushing a nupkg by filename to a Symbol Server with no matching snupkg, do not show a File Not Found error. 
+        /// </summary>
+        [Fact]
+        public void PushCommand_Server_Nupkg_FilenameSnupkgDoesNotExist_NoFileNotFoundError()
+        {
+            // Arrange
+            using (var packageDirectory = TestDirectory.Create())
+            {
+                var nuget = Util.GetNuGetExePath();
+
+                string packageId = "packageWithoutSnupkg";
+                string version = "1.1.0";
+
+                //Create Nupkg in test directory.
+                string nupkgFullPath = Util.CreateTestPackage(packageId, version, packageDirectory);
+
+                string nupkgFileName = Util.BuildPackageString(packageId, version, NuGetConstants.PackageExtension);
+                string snupkgFileName = Util.BuildPackageString(packageId, version, NuGetConstants.SnupkgExtension);
+                string snupkgFullPath = Path.Combine(packageDirectory, snupkgFileName);
+
+                using (var server = CreateAndStartMockV3Server(packageDirectory, out string sourceName))
+                {
+                    SetupMockServerAlwaysCreate(server);
+
+                    // Act
+                    var result = CommandRunner.Run(
+                        nuget,
+                        packageDirectory,
+                        $"push {nupkgFullPath} -Source {sourceName} -Timeout 110",
+                        waitForExit: true,
+                        timeOutInMilliseconds: 120000); // 120 seconds
+
+                    // Assert
+
+                    //Ignoring filename in File Not Found error since the error should not appear in any case.
+                    string genericFileNotFoundError = string.Format(MESSAGE_FILE_DOES_NOT_EXIST, string.Empty);
+
+                    Assert.True(0 == result.Item1, "Expected to successfully push a nupkg without a snupkg.");
+                    Assert.Contains(MESSAGE_PACKAGE_PUSHED, result.Item2);
+                    Assert.DoesNotContain(genericFileNotFoundError, result.Item3);
+                }
+            }
+        }
+
+        /// <summary>
+        /// When pushing *.nupkg to a symbol server, but no snupkgs are selected with that wildcard, there is (TODO: not) a FileNotFound error about snupkgs.
+        /// TODO: bug fixes will come from https://github.com/NuGet/Home/issues/8148
+        /// </summary>
+        [Fact]
+        public void PushCommand_Server_Nupkg_WildcardSnupkgDoesNotExist_FileNotFoundError()
```

Comment:
How do you explicitly push to a symbol server? 

Isn't this the same behavior for any globbing pattern?

Reply:
The server needs to have [SymbolPackagePublish](https://github.com/NuGet/NuGet.Client/blob/f256c56155a5b32d54ef3d8cf2bc28d197862df4/test/NuGet.Clients.Tests/NuGet.CommandLine.Test/Util.cs#L578) as I setup the Mock server to do.

Reply:
And no, the behavior is different for globbing snupkgs vs globbing nupkgs. Which is the point of many of these new tests.

Reply:
Client behavior is to try to push the equivalent snupkg of whatever nupkg was pushed. 
*.nupkg -> *.snupkg
filename.nupkg -> filename.snupkg

No, *.nupkg is an Error, and it's  not explicitly tested. `PushCommand_Server_Nupkg_WildcardSnupkgDoesNotExist_FileNotFoundError`. was intended to test *.nupkg where it finds nupkgs, but no snupkgs. When I refactor, I'll clear this one up. Thanks for pointing it out.

Reply:
So this is testing the buggy behavior right now?

Reply:
These tests are testing existing behavior. We have determined that it's incorrect behavior, so I will be fixing it in another Issue.

Reply:
Regarding the *.nupkg not found, I added 2 additional tests for that.